### PR TITLE
Pytest style analysis/test_distances.py

### PR DESCRIPTION
**Pull request by utkbansal**
Fixes #

Changes made in this Pull Request:
 - 

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

File: testsuite/MDAnalysisTests/analysis/test_distances.py [hunk 1]
```diff
@@ -21,91 +21,71 @@
...
+                        [1, 0, 1, 1, 1]], dtype=np.bool)
+
 
     def test_np(self):
-        contacts = MDAnalysis.analysis.distances.contact_matrix(
-            self.coord, cutoff=1, returntype="numpy")
-        assert_equal(contacts.shape, self.shape,
-                     "wrong shape (should be {0})".format(self.shape))
+        contacts = MDAnalysis.analysis.distances.contact_matrix(self.coord, cutoff=1, returntype="numpy")
+        assert contacts.shape == self.shape, "wrong shape (should be {0})".format(self.shape)
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/analysis/test_distances.py]: This looks like you break the 80 char line limit. Also these tests look like they could be parametrized.

Comment (utkbansal) [inline testsuite/MDAnalysisTests/analysis/test_distances.py]: @kain88-de All the 4 methods have different check in the `assert_equal` statement. If you want to parametrize, I'll have to send both the values that will be compared. I don't think that's a good idea.

Comment (kain88-de) [inline testsuite/MDAnalysisTests/analysis/test_distances.py]: called `np.asarray(...)` on the contacts should also work for the sparse case

File: testsuite/MDAnalysisTests/analysis/test_distances.py [hunk 2]
```diff
@@ -21,91 +21,84 @@
...
-                                       [0, 0, 0, 1, 0],
-                                       [0, 0, 0, 0, 1]], dtype=np.bool)
-        self.res_pbc = np.array([[1, 0, 1, 1, 1],
-                                    [0, 1, 0, 0, 0],
-                                    [1, 0, 1, 1, 1],
-                                    [1, 0, 1, 1, 1],
-                                    [1, 0, 1, 1, 1]], dtype=np.bool)
+
+class TestContactMatrix(object):
+    coord = np.array([[1, 1, 1],
```
Comment (richardjgowers) [inline testsuite/MDAnalysisTests/analysis/test_distances.py]: The problem with declaring arrays in the class namespace like this is they can be accidentally modified by a test, and they will remain changed for all other tests (I think?)

File: testsuite/MDAnalysisTests/analysis/test_distances.py [hunk 3]
```diff
@@ -21,91 +21,84 @@
...
-
-    def tearDown(self):
-        del self.u
-        del self.ag
-        del self.u2
-        del self.ag2
-        del self.expected
+
+class TestDist(object):
+    u = MDAnalysis.Universe(GRO)
```
Comment (richardjgowers) [inline testsuite/MDAnalysisTests/analysis/test_distances.py]: Same here, we shouldn't just be loading everything into the class namespace

**Comment by utkbansal** [comment]
@richardjgowers Better now?

**Comment by kain88-de** [comment]
@utkbansal the coverage went down with the latest changes

**Comment by utkbansal** [comment]
@kain88-de I don't see how that happened, have restarted the build.

**Pull request merged**